tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_opcodes.py

```diff
@@ -375,259 +416,1614 @@ def test_bal_extcodesize_and_oog(
-    "fails_at_call", [True, False], ids=["oog_at_call", "successful_call"]
+    "target_is_warm", [False, True], ids=["cold_target", "warm_target"]
+)
+@pytest.mark.parametrize(
+    "target_is_empty", [False, True], ids=["existing_target", "empty_target"]
+)
+@pytest.mark.parametrize("value", [0, 1], ids=["no_value", "with_value"])
+@pytest.mark.parametrize(
+    "memory_expansion", [False, True], ids=["no_memory", "with_memory"]
 )
-def test_bal_call_and_oog(
+def test_bal_call_success_no_delegation(
     pre: Alloc,
     blockchain_test: BlockchainTestFiller,
     fork: Fork,
-    fails_at_call: bool,
+    target_is_warm: bool,
+    target_is_empty: bool,
+    value: int,
+    memory_expansion: bool,
+    call_warmup_bytecode_cost: int,
 ) -> None:
-    """Ensure BAL handles CALL and OOG during CALL appropriately."""
-    alice = pre.fund_eoa()
-    bob = pre.fund_eoa()
+    """CALL without 7702 delegation - test successful execution."""
     gas_costs = fork.gas_costs()
+    alice = pre.fund_eoa()
 
-    # Create contract that attempts to call Bob
-    call_contract_code = Bytecode(
-        Op.PUSH1(0)  # retSize
-        + Op.PUSH1(0)  # retOffset
-        + Op.PUSH1(0)  # argsSize
-        + Op.PUSH1(0)  # argsOffset
-        + Op.PUSH1(0)  # value
-        + Op.PUSH20(bob)  # address
-        + Op.PUSH2(0xFFFF)  # gas (provide enough for the call)
-        + Op.CALL  # Call (cold account access)
-        + Op.STOP
+    target = (
+        pre.empty_account()
+        if target_is_empty
+        else pre.deploy_contract(code=Op.STOP)
     )
 
-    call_contract = pre.deploy_contract(code=call_contract_code)
+    # warmup code if needed
+    warmup_code = Bytecode()
+    if target_is_warm:
+        warmup_code = Op.POP(Op.EXTCODESIZE(target))
 
-    intrinsic_gas_calculator = fork.transaction_intrinsic_cost_calculator()
-    intrinsic_gas_cost = intrinsic_gas_calculator()
+    # memory expansion / no expansion
+    ret_size = 32 if memory_expansion else 0
 
-    # Costs:
-    # - 7 PUSH operations = G_VERY_LOW * 7
-    # - CALL cold = G_COLD_ACCOUNT_ACCESS (minimum for account access)
-    push_cost = gas_costs.G_VERY_LOW * 7
-    call_cold_cost = gas_costs.G_COLD_ACCOUNT_ACCESS
-    tx_gas_limit = intrinsic_gas_cost + push_cost + call_cold_cost
-
-    if fails_at_call:
-        # subtract 1 gas to ensure OOG at CALL
-        tx_gas_limit -= 1
+    # caller contract
+    call_code = warmup_code + Op.CALL(
+        gas=0, address=target, value=value, ret_size=ret_size
+    )
+    caller = pre.deploy_contract(code=call_code, balance=value)
 
-    tx = Transaction(
-        sender=alice,
-        to=call_contract,
-        gas_limit=tx_gas_limit,
+    intrinsic_cost = fork.transaction_intrinsic_cost_calculator()()
+
+    # Warmup cost: bytecode (PUSH20 + POP) + state access (EXTCODESIZE cold)
+    warmup_total = (
+        (call_warmup_bytecode_cost + gas_costs.G_COLD_ACCOUNT_ACCESS)
+        if target_is_warm
+        else 0
```

Comment:
Does this include the base cost? Even if the account is warm, CALLs still cost 100 gas (where is this included?)